actionpack/lib/action_controller/metal/head.rb

```diff
@@ -17,8 +17,14 @@ module Head
     #
     # See Rack::Utils::SYMBOL_TO_STATUS_CODE for a full list of valid +status+ symbols.
     def head(status, options = {})
-      options, status = status, nil if status.is_a?(Hash)
-      status ||= options.delete(:status) || :ok
+      if status.is_a?(Hash)
+        msg = "Passing Hash as first parameter"
+        options, status = status, nil
+        status ||= options.delete(:status) || (
+          (msg += " and the implicit :ok status") && :ok
+        )
+        ActiveSupport::Deprecation.warn(msg + " for `head` has been deprecated and will be removed in Rails 5.1.")
+      end
```

Comment:
Just a minor nit as well, [Hash#delete] accepts a block, which can be used to set defaults:

E.g. `status.delete(:status) { :ok }`